pyproject.toml

```diff
@@ -4,12 +4,12 @@ requires = [
     "setuptools",
     "wheel",
     "Cython>=0.28.5",
-    "numpy==1.13.3; python_version=='3.6' and platform_system!='AIX' and platform_python_implementation == 'CPython'",
-    "numpy==1.14.0; python_version=='3.6' and platform_system!='AIX' and platform_python_implementation != 'CPython'",
-    "numpy==1.14.5; python_version=='3.7' and platform_system!='AIX'",
-    "numpy==1.17.3; python_version>='3.8' and platform_system!='AIX'",
-    "numpy==1.16.0; python_version=='3.6' and platform_system=='AIX'",
-    "numpy==1.16.0; python_version=='3.7' and platform_system=='AIX'",
-    "numpy==1.17.3; python_version>='3.8' and platform_system=='AIX'",
+    # We follow scipy for much of these pinnings
+    # https://github.com/scipy/scipy/blob/master/pyproject.toml
+    "numpy==1.16.5; python_version=='3.6'",
+    "numpy==1.16.5; python_version=='3.7'",
+    "numpy==1.17.3; python_version=='3.8'",
+    # do not pin numpy on future versions of python to avoid incompatible numpy and python versions
+    "numpy; python_version>='3.9'",
```

Comment:
We need to pin the oldest version of numpy that supports python_version == '3.9' because it will define the ABI compatibility expectations for the scikit-learn wheel.

Reply:
I'm not sure I follow this comment. This is the same as scipy, and I figured that would be acceptable. I'm not sure if you can explain your thinking a little more.

I *can* change it to 1.19.3 if needed, since that is the first version that has released wheels for Python 3.9. I just figured it might cut down on the maintenance burden and I assumed that was why scipy set that as their default.

Reply:
We use the version of numpy pyproject.toml to build the wheels in a github actions workflow. scikit-learn has Cython extension that need numpy at build time and the generated wheel need to be built with the oldest supported version of numpy to make sure that our build do not depend on recently changes of the numpy ABI.

For instance the future scikit-learn 0.24 wheel for python 3.8 needs to be build with numpy 1.17.3 to make  sure that users who have such old versions of numpy can install a binary package of scikit-learn that is binary compatible.

Reply:
The wheel building workflow is here:

https://github.com/scikit-learn/scikit-learn/blob/master/.github/workflows/wheels.yml

Reply:
@ogrisel yeah, I gathered as much as you explained. Not sure why scipy chose to not pin numpy the same way, but I'm fine with setting this to 1.19.3, and thanks for the pointer to the new workflow. cibuildwheels looks nice :slightly_smiling_face: 

@alfaro96 yeah I got that :wink: , but thanks for making it explicit